client/components/overview/StyleSelector.jsx

```diff
@@ -1,23 +1,38 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { useEffect } from 'react';
 import map from 'lodash/map';
+import PropTypes from 'prop-types';
 import StyleButton from './StyleButton';
 import { StyleSelectorContainer } from './styles/StyleSelector.style';
 
-const StyleSelector = ({ styles }) => (
-  <StyleSelectorContainer
-    data-testid="style-selector"
-    className="style-selector"
-  >
-    {styles.results.map((style) => (
-      map(style.photos, (photo) => (
-        <StyleButton
-          photo={photo}
-        />
-      ))
-    ))}
-  </StyleSelectorContainer>
-);
+const StyleSelector = ({ productStyles, currentStyle, setCurrentStyle }) => {
+  useEffect(() => {
```

Comment:
I think you're on the right track w/ the useEffect hook watching on changes in `productStyles` and `currentStyle`. I'm not sure if this is the right component to add this in though. It might be a good idea to first make separate hooks for each of your dependencies & only add them to the same useEffect if the side effects needed end up being the same.